homeassistant/components/mqtt_room/sensor.py

```diff
@@ -74,36 +80,51 @@ async def async_setup_platform(
                 config.get(CONF_DEVICE_ID),
                 config.get(CONF_TIMEOUT),
                 config.get(CONF_AWAY_TIMEOUT),
+                config.get(CONF_EXTRA_ROOMS),
             )
         ]
     )
 
 
+class MQTTRoomState(TypedDict):
+    """Hold the state of a room."""
+
+    room: str
+    distance: float
+    updated: datetime
+
+
 class MQTTRoomSensor(SensorEntity):
     """Representation of a room sensor that is updated via MQTT."""
 
-    def __init__(self, name, state_topic, device_id, timeout, consider_home):
+    def __init__(
+        self, name, state_topic, device_id, timeout, consider_home, extra_rooms
+    ):
         """Initialize the sensor."""
-        self._state = STATE_NOT_HOME
         self._name = name
         self._state_topic = f"{state_topic}/+"
         self._device_id = slugify(device_id).upper()
         self._timeout = timeout
         self._consider_home = (
             timedelta(seconds=consider_home) if consider_home else None
         )
-        self._distance = None
-        self._updated = None
+        self._room_states: list[MQTTRoomState] = [
+            MQTTRoomState(room=STATE_NOT_HOME, distance=None, updated=None),
+        ]
+
+        for _ in range(extra_rooms):
```

Comment:
@jbouwh CONF_EXTRA_ROOMS is (ultimately) being used here.

Reply:
Okay, I think we also should use it when processing an incoming payload.